pathio.go

```diff
@@ -140,3 +142,30 @@ func getRegionForBucket(name string) (string, error) {
 	}
 	return *resp.LocationConstraint, nil
 }
+
+type s3Handler interface {
+	GetBucketLocation(input *s3.GetBucketLocationInput) (*s3.GetBucketLocationOutput, error)
+	GetObject(input *s3.GetObjectInput) (*s3.GetObjectOutput, error)
+	PutObject(input *s3.PutObjectInput) (*s3.PutObjectOutput, error)
+}
+
+type liveS3Handler struct {
```

Comment:
<nit> I'm not sure `live` is the best naming convention for this. I think we've gone with  `s3HandlerImpl`